### PR TITLE
[libc] add headers.txt for windows

### PR DESCRIPTION
**Pull request by nickdesaulniers**
Link: https://github.com/llvm/llvm-project/pull/117220#issuecomment-2518126598

### CODE REVIEW

**Comment by nickdesaulniers** [comment]
@SchrodingerZhu if you have a windows build handy, can you please help test this out? This would be doing a fullbuild on windows.

**Comment by SchrodingerZhu** [comment]
> @SchrodingerZhu if you have a windows build handy, can you please help test this out? This would be doing a fullbuild on windows.

trying out now

**Comment by SchrodingerZhu** [comment]
No issue found running `ninja check-libc`

**Review by SchrodingerZhu — APPROVED** [review]

**Comment by ilovepi** [comment]
I also kicked off https://ci.chromium.org/ui/p/fuchsia/builders/toolchain.ci.shadow/clang-windows-x64/b8729459123378281649/overview with this patch.

File: libc/config/windows/headers.txt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    libc.include.assert
+    libc.include.ctype
+    libc.include.dirent
+    libc.include.errno
+    libc.include.features
+    libc.include.fenv
+    libc.include.float
+    libc.include.inttypes
+    libc.include.limits
+    libc.include.link
```
Comment (nickdesaulniers) [inline libc/config/windows/headers.txt:11]: probably don't want the linux specific link.h for windows.  I think the rest are fine.

File: libc/config/windows/headers.txt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    libc.include.stdint
+    libc.include.stdio
+    libc.include.stdlib
+    libc.include.string
+    libc.include.strings
+    libc.include.threads
+    libc.include.time
+    libc.include.uchar
+    libc.include.unistd
+    libc.include.wchar
```
Comment (michaelrj-google) [inline libc/config/windows/headers.txt:30]: I'd recommend limiting this to only headers which have functions enabled on windows:
```suggestion
    libc.include.ctype
    libc.include.string
    libc.include.inttypes
    libc.include.stdlib
    libc.include.errno
    libc.include.fenv
    libc.include.math
```

**Review by michaelrj-google — APPROVED** [review]

**Comment by nickdesaulniers** [comment]
> I also kicked off https://ci.chromium.org/ui/p/fuchsia/builders/toolchain.ci.shadow/clang-windows-x64/b8729459123378281649/overview with this patch.

Looks like that passed to me, but its also my first time looking at that such dashboard.

**Pull request merged**